### PR TITLE
docs: add docs about integration

### PR DESCRIPTION
**Pull request by claudemamo**

### CODE REVIEW

**Comment by claudemamo** [comment]
@amcgee @Topener : do you think we can get someone from the design team to draw us an integration icon for the front page? The one there is now isn't the best illustration: 

image

File: docs/integration/camel-dhis2-component.md [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+}
+```
+
+:::note
+Visit the [DHIS2 Java SDK page on the DHIS2 Developer Portal](/docs/integration/dhis2-java-sdk) to learn more about the DHIS2 Java SDK.
+:::
+
+The `client` parameter in the above route snippet is set to `#dhis2Client`. `#dhis2Client` is an entry in Camel registry which is bound to an instance of `Dhis2Client` thanks to the following line of code:
+
+`getCamelContext().getRegistry().bind("dhis2Client", dhis2Client);`
```
Comment (Topener) [inline docs/integration/camel-dhis2-component.md:76]: There are several blocks in this file that are with single-ticks, instead of code-blocks. Could you convert them all to code-blocks? 

Also in all the endpoints/methods a lot of code-blocks that are with single ticks

File: src/pages/index.js
Comment (Topener) [inline src/pages/index.js:1]: Can you keep this in a separate PR with a `branch` on the `dhis2/developer-portal` repo? We'll need to add more blocks here since the docs merge, so best to do that at once.

File: docs/integration/apache-camel.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+        .to( "dhis2://get/collection?path=organisationUnits&arrayName=organisationUnits&client=#dhis2Client")
+        .marshal().csv();
+  }
+}
+```
+
+The route is defined within the `configure()` method. `from(...)` and `to(...)` are endpoints while `marshal()` and `json()` are processors. The `HelloWorldRouteBuilder` class consists only of a few lines of code, yet, it is punching well above its weight. The route has a [`servlet` endpoint](https://camel.apache.org/components/4.0.x/servlet-component.html) listening for HTTP requests to trigger the rest of the route's execution. On receiving a HTTP request, the route fetches organisation units from a DHIS2 server using the DHIS2 endpoint. The route then serialises the received organisation units to CSV before including them in the response to the HTTP client which kicked off the route. 
+
+:::note
+Visit the [Camel DHIS2 Component page on the DHIS2 Developer Portal](/docs/integration/camel-dhis2-component) to learn more about DHIS2 endpoints.
```
Comment (Topener) [inline docs/integration/apache-camel.md:65]:
```suggestion
Visit the [Camel DHIS2 Component page](/docs/integration/camel-dhis2-component) to learn more about DHIS2 endpoints.
```
"On the developer portal" seems not needed as the reader is already there

File: docs/integration/camel-dhis2-component.md [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+        getCamelContext().getRegistry().bind("dhis2Client", dhis2Client);
+
+        from("direct:resourceTablesAnalytics")
+            .to("dhis2://resourceTables/analytics?skipAggregate=true&skipEvents=true&lastYears=1&client=#dhis2Client");
+    }
+}
+```
+
+:::note
+Visit the [DHIS2 Java SDK page on the DHIS2 Developer Portal](/docs/integration/dhis2-java-sdk) to learn more about the DHIS2 Java SDK.
```
Comment (Topener) [inline docs/integration/camel-dhis2-component.md:71]:
```suggestion
Visit the [DHIS2 Java SDK page](/docs/integration/dhis2-java-sdk) to learn more about the DHIS2 Java SDK.
```

File: docs/integration/camel-dhis2-component.md [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+
+`dhis2://resourceTables/analytics` does not produce a new message body. The following table describes the parameters specific to `dhis2://resourceTables/analytics`:
+
+| **Parameter Name** |                              **Description**                             |
+|:------------------:|:------------------------------------------------------------------------:|
+|    skipAggregate   | Skip generation of aggregate data and completeness data.                 |
+|     skipEvents     | Skip generation of event data.                                           |
+|      lastYears     | Number of last years of data to include.                                 |
+|      interval      | Duration in milliseconds between completeness checks. Default is `30000` |
+|        async       | Whether to block until analytics is complete. Default is `false`.        |
```
Comment (Topener) [inline docs/integration/camel-dhis2-component.md:303]: Maybe add a column specifically for default values? Also automatically explains what happens with the first 3 parameters when not supplied.

File: docs/integration/camel-dhis2-component.md [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+    <artifactId>camel-dhis2</artifactId>
+    <version>x.x.x</version>
+    <!-- use the same version as your Camel core version -->
+</dependency>
+```
+
+## Getting started
+
+:::note
+A basic understanding of Apache Camel and endpoints is assumed. Visit the [Apache Camel page on the DHIS2 Developer Portal](/docs/integration/apache-camel) to learn the fundamental Camel concepts. All Camel route examples given in this page are expressed in the Java-based DSL.
```
Comment (Topener) [inline docs/integration/camel-dhis2-component.md:21]:
```suggestion
A basic understanding of Apache Camel and endpoints is assumed. Visit the [Apache Camel page](/docs/integration/apache-camel) to learn the fundamental Camel concepts. All Camel route examples given in this page are expressed in the Java-based DSL.
```

File: docs/assets/integration/api-vs-db-integration.excalidraw
Comment (Topener) [inline docs/assets/integration/api-vs-db-integration.excalidraw:1]: Is this needed anywhere?

File: docs/assets/integration/camel-concepts.excalidraw
Comment (Topener) [inline docs/assets/integration/camel-concepts.excalidraw:1]: Is this file needed?

Comment (Topener) [reply docs/assets/integration/camel-concepts.excalidraw:1]: This comment also applies to the other `excalidraw` files

**Review by Topener — CHANGES_REQUESTED** [review]
A few minor changes requested, but other than that pretty good.

File: docs/assets/integration/api-vs-db-integration.excalidraw
Comment (claudemamo) [reply docs/assets/integration/api-vs-db-integration.excalidraw:1]: This is the source code of the diagrams. This is needed so that we can edit the diagrams in the future.

File: docs/integration/camel-dhis2-component.md (hunk 4 above)
Comment (claudemamo) [reply docs/integration/camel-dhis2-component.md:323]: Frankly, no idea what happens if these settings are omitted. The defaults are not documented in the DHIS2 API docs.

File: docs/assets/integration/api-vs-db-integration.excalidraw
Comment (Topener) [reply docs/assets/integration/api-vs-db-integration.excalidraw:1]: Ah I see, thats alright. We might want to look into excluding them from the build but it has no harm for now, I will put that on my todo list

**Review by Topener — APPROVED** [review]
LGTM

**Comment by claudemamo** [comment]
> LGTM

All that's left is for @cooper-joe to provide us with an appropriate integration icon. The current one was lifted from somewhere else.
